res/css/views/rooms/_MessageComposer.scss

```diff
@@ -63,6 +63,22 @@ limitations under the License.
     .mx_VoiceMessagePrimaryContainer {
         margin-right: $spacing-8;
     }
+
+    .mx_ReplyPreview_header {
+        overflow: hidden;
+
+        .mx_ReplyPreview_header_replyTo {
+            // 16px: cancel button
+            max-width: calc(100% - var(--ReplyPreview_padding-inline) - 16px - var(--ReplyPreview_gap-inline));
```

Comment:
I believe we should change the `position: absolute` of `mx_ReplyPreview_header_cancel` so that it's in the flow of the page, and we can add `flex: 1` to the text container that needs to be truncated

The big max width calculation will be offloaded to the browser

Reply:
Should it be scope of this PR?

Reply:
I think this is an alternative approach to how you're doing it, that moves the heavy lifting of width calculation to the browser. Sounds like something we'd aspire to yes

Reply:
Please note that the same kind of approach has been in effect on BaseCard, for example TimelineCard's header.